pkg/controller/instance/instance_controller.go

```diff
@@ -382,22 +386,40 @@ func GetOperatorVersion(instance *kudov1beta1.Instance, c client.Client) (ov *ku
 	return ov, nil
 }
 
-// paramsMap generates {{ Params.* }} map of keys and values which is later used during template rendering.
-func paramsMap(instance *kudov1beta1.Instance, operatorVersion *kudov1beta1.OperatorVersion) map[string]string {
-	params := make(map[string]string)
-
-	for k, v := range instance.Spec.Parameters {
-		params[k] = v
+func paramValue(v *string, t v1beta1.ParameterValueType) (r interface{}, err error) {
+	switch t {
+	case kudov1beta1.DictValueType:
+		r, err = kudo.YAMLDict(kudo.StringValue(v))
```

Comment:
Seems like we need a yaml encoder... this is NOT a kudo thing.... it also looks like `kudo` is growing to be a util package (which I hope it doesn't)

Reply:
It's already too late, this is `pkg/util/kudo` 😄. But given that this package only consists of `convert_types.go`, I'll rename this package to something different than `kudo` to make clear that it's about type conversions.